sospice/catalog/catalog.py

```diff
@@ -451,3 +453,53 @@ def plot_fov(self, ax, **kwargs):
                 fontsize="xx-large",
                 borderaxespad=2,
             )
+
+    def download_files(
+        self,
+        base_dir,
+        base_url=None,
+        keep_tree=True,
+        downloader=None,
+        max_download=1000,
+    ):
+        """
+        Download all files from Catalog.,
+
+        Parameters
+        ----------
+        base_dir: Path or str
+            Base directory to download file to
+        base_url: str
+            Base URL for file
+        keep_tree: bool
+            Keep tree directory structure (by level and date)
+        downloader: parfive.Downloader
+            If provided, enqueue file for download instead of downloading it.
+            To download enqueued files, run `downloader.download()`
+        max_download: int
+            default maximum of 1000 files can be downloaded.
+            User can override it by changing the value.
+
+        Return
+        ------
+        parfive.Result
+            Download result (or None if file has only been enqueued)
+        """
+        get_catalog = self.read_catalog()
+        if max_download > 1000:
+            warnings.warn(
+                "You are overriding the default max_download: This might cause performance issues.",
+                UserWarning,
+            )
+
```

Comment:
I would put here the creation of the `Download` object if not provided: it is better to create only one such object, instead of creating one every time `FileMetadata.download_file()` is called. Furthermore, this is the only way to take advantage of the parallel downloading of the files queued in the `Download` object.

Reply:
So in the end there *will* be some duplicated code with `FileMetadata.download_file()` (everything linked to the `downloader is None` and `do_download` conditions), but this is not really a problem (unless you find a better solution).